File: 프로그래머스/2/84512. 모음 사전/모음 사전.py
def solution(word):
    count = 0
    answer = 0
    
    def check(now_word, max_len, word):
        nonlocal count
        nonlocal answer
        
        if now_word == word:
            answer = count
            return
        
        if max_len <= len(now_word):
            return
        
        alpha = ['A', 'E', 'I', 'O', 'U']
        for a in alpha:
            count += 1
            check(now_word + a, max_len, word)
        
    
    check("", 5, word)
    return answer

File: 프로그래머스/2/84512. 모음 사전/모음 사전.java
import java.util.*;

class Solution {
    int count = 0;
    int answer = 0;
    
    public void check(String nowWord, String word) {
        if (word.equals(nowWord)) {
            answer = count;
            return;
        } 
        
        if (nowWord.length() >= 5) return;
        
        String[] alpha = {"A", "E", "I", "O", "U"};
        for (String a: alpha) {
            count += 1;
            check(nowWord + a, word);
        }
    }
    
    public int solution(String word) {
        check("", word);
        return answer;
    }
}